pyracf/common/security_admin.py

```diff
@@ -64,25 +69,47 @@ def __init__(
         self._trait_map = {}
         self.__debug = debug
         self.__generate_requests_only = generate_requests_only
-        if add_field_data is not None:
-            self.__add_field_data(add_field_data)
-        if overwrite_field_data is not None:
-            self.__overwrite_field_data(overwrite_field_data)
+        if update_existing_segment_traits is not None:
+            self.__add_additional_valid_segment_traits(update_existing_segment_traits)
+        if overwrite_existing_segment_traits is not None:
+            self.__overwrite_valid_segment_traits(overwrite_existing_segment_traits)
+        if additional_secret_traits is not None:
+            self.__add_additional_secret_traits(additional_secret_traits)
 
     # ============================================================================
     # Custom Fields
     # ============================================================================
-    def __add_field_data(self, field_data: dict):
+    def __add_additional_valid_segment_traits(
```

Comment:
Rename this to `__update_valid_segment_traits` because we are not simply appending, we are effectively overlaying the existing segment traits. Similarly update the local variable names.

Reply:
agree; change made